sql/core/src/main/scala/org/apache/spark/sql/execution/window/BoundOrdering.scala

```diff
@@ -25,18 +25,22 @@ import org.apache.spark.sql.catalyst.expressions.Projection
  * Function for comparing boundary values.
  */
 private[window] abstract class BoundOrdering {
-  def compare(inputRow: InternalRow, inputIndex: Int, outputRow: InternalRow, outputIndex: Int): Int
+  def compare(
+      inputRow: InternalRow,
+      inputIndex: Long,
+      outputRow: InternalRow,
+      outputIndex: Long): Long
 }
 
 /**
  * Compare the input index to the bound of the output index.
  */
-private[window] final case class RowBoundOrdering(offset: Int) extends BoundOrdering {
+private[window] final case class RowBoundOrdering(offset: Long) extends BoundOrdering {
```

Comment:
It does not make any sense to make the offsets longs. This is an execution detail, buffer indexes (which are integer bound), and you really should not be messing with those.

Try to keep your change more local, and only modify `WindowExec.createBoundOrdering` and the code generating the `WindowExec.windowFrameExpressionFactoryPairs`. That should be enough.